java/src/main/kotlin/io/spine/protodata/java/MethodCalls.kt

```diff
@@ -0,0 +1,126 @@
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.protodata.java
+
+import com.google.common.collect.ImmutableList
+import com.google.common.collect.ImmutableMap
+import com.google.protobuf.Message
+
+internal const val OF = "of"
+
+/**
+ * Constructs a call to a static method of this class.
+ *
+ * @param name The name of the method.
+ * @param arguments The method arguments.
+ * @param generics The method type parameters.
+ */
+@JvmOverloads
+public fun <T> ClassName.call(
+    name: String,
+    arguments: List<Expression<*>> = listOf(),
+    generics: List<ClassName> = listOf()
+): MethodCall<T> = MethodCall(this, name, arguments, generics)
+
+/**
+ * Constructs an expression which creates a new builder for this class.
+ *
+ * Example: `ClassName("com.acme.Bird").newBuilder()` yields
+ * `"com.acme.Bird.newBuilder()"`.
+ */
+public fun ClassName.newBuilder(): MethodCall<Message.Builder> =
+    MethodCall(this, "newBuilder")
+
+/**
+ * Constructs an expression which obtains the default instance for this class.
+ *
+ * Example: `ClassName("com.acme.Bird").getDefaultInstance()` yields
+ * `"com.acme.Bird.getDefaultInstance()"`.
+ */
+public fun ClassName.getDefaultInstance(): MethodCall<Message> =
+    call("getDefaultInstance")
+
+/**
+ * Constructs an expression that returns the Protobuf enum value by
+ * the given number from this class.
+ *
+ * Example: `ClassName("com.acme.Bird").enumValue(1)` yields
+ * `"com.acme.Bird.forNumber(1)"`.
+ */
+public fun EnumName.enumValue(number: Int): MethodCall<Message> =
+    call("forNumber", listOf(Literal(number)))
+
+/**
+ * Constructs an expression of a list from the given list of [expressions].
+ *
+ * The resulting expression always yields an instance of Guava `ImmutableList`.
+ */
+public fun listExpression(expressions: List<Expression<*>>): MethodCall<ImmutableList<*>> =
+    ClassName(ImmutableList::class).call(OF, expressions)
+
+/**
+ * Constructs an expression of a list of the given [expressions].
+ *
+ * The resulting expression always yields an instance of Guava `ImmutableList`.
+ */
+public fun listExpression(vararg expressions: Expression<*>): MethodCall<ImmutableList<*>> =
+    listExpression(expressions.toList())
+
+/**
+ * Constructs an expression of a map of the given [expressions].
+ *
+ * The resulting expression always yields an instance of Guava `ImmutableMap`.
+ *
+ * @param expressions The expressions representing the entries.
+ * @param keyType The type of the keys in the map;
+ *   must be non-`null` if the map is not empty, may be `null` otherwise.
+ * @param valueType The type of the values in the map;
+ *   must be non-`null` if the map is not empty, may be `null` otherwise.
+ */
+public fun mapExpression(
```

Comment:
Can we have two overloads for this function? One would be without parameters representing construction of an empty map. The other one would be with non-empty parameter for `expressions` and non-`null` types for keys and values. 

This way we would not need to have an awkward check for non-`null` in nullable parameters.